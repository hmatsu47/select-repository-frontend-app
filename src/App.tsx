import { Component } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import { TitleBar } from "./TitleBar";

export const App: Component = () => {
  return (
    <>
      <TitleBar />
      <Box
        sx={{
          padding: "10px 0 0 0",
          width: "100%",
          minWidth: "960px",
          display: "flex",
          justifyContent: "center",
        }}
        aria-live="polite"
      >
        <Stack spacing={2} direction="column"></Stack>
      </Box>
    </>
  );
};
