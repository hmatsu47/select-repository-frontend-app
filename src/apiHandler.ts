import { fetchWithTimeout } from "./fetchWithTimeout";

// APIからデータ取得（GET）
export const getApiData = async (url: string) => {
  return await (await fetchWithTimeout(url)).json();
};
// APIからデータ取得（POST）
export const postApiData = async (url: string, object: Object) => {
  return await (await fetchWithTimeout(url, object)).json();
};
