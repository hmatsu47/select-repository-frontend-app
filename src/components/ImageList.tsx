import { For, Show } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Paper from "@suid/material/Paper";
import Table from "@suid/material/Table";
import TableBody from "@suid/material/TableBody";
import TableCell from "@suid/material/TableCell";
import TableContainer from "@suid/material/TableContainer";
import TableHead from "@suid/material/TableHead";
import TableRow from "@suid/material/TableRow";
import Typography from "@suid/material/Typography";
import {
  images,
  imageUri,
  isOpenedConfirm,
  repository,
  repositoryUri,
  setImageUri,
} from "../signal";
import { formatDateTimeDisplay } from "../formatDateTime";

export const ImageList = () => {
  return (
    <Show
      when={repository() && repositoryUri() && images() !== undefined}
      fallback={<></>}
    >
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <Show
          when={images() !== null}
          fallback={
            <Typography variant="h6">
              リポジトリにイメージがありません
            </Typography>
          }
        >
          <TableContainer component={Paper}>
            <Table aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell>選択</TableCell>
                  <TableCell>
                    イメージURI
                    <br />
                    （リポジトリURI:タグ または リポジトリURI@ダイジェスト）
                  </TableCell>
                  <TableCell>プッシュ日時</TableCell>
                  <TableCell>
                    サイズ
                    <br />
                    （MB）
                  </TableCell>
                  <TableCell>
                    脆弱性調査
                    <br />
                    （要AWSログイン）
                  </TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                <For each={images()} fallback={<></>}>
                  {(imageItem) => (
                    <TableRow>
                      <TableCell component="th" scope="row">
                        <Button
                          disabled={isOpenedConfirm()}
                          variant={
                            imageUri() && imageUri() === imageItem.uri
                              ? "contained"
                              : "outlined"
                          }
                          size="small"
                          color={
                            imageUri() && imageUri() === imageItem.uri
                              ? "primary"
                              : "inherit"
                          }
                          onClick={() => {
                            setImageUri(imageItem.uri);
                          }}
                          sx={{ textTransform: "none" }}
                          title={`${imageItem.uri} を選択`}
                        >
                          選択
                        </Button>
                      </TableCell>
                      <TableCell>{imageItem.uri}</TableCell>
                      <TableCell>
                        {formatDateTimeDisplay(new Date(imageItem.pushed_at))}
                      </TableCell>
                      <TableCell align="right">
                        {(
                          Math.round(
                            (imageItem.size / (1000 * 1000)) * Math.pow(10, 2)
                          ) / Math.pow(10, 2)
                        ).toLocaleString()}
                      </TableCell>
                      <TableCell>
                        <Button
                          disabled={isOpenedConfirm()}
                          variant="text"
                          onClick={() => {
                            const url = `https://${
                              repositoryUri()!.split(".")[3]
                            }.console.aws.amazon.com/ecr/repositories/private/${
                              repositoryUri()!.split(".")[0]
                            }/${repository()}/image/${
                              imageItem.digest
                            }/scan-results/?region=${
                              repositoryUri()!.split(".")[3]
                            }`;
                            window.open(url, "_blank");
                          }}
                          sx={{ textTransform: "none" }}
                          title={`${imageItem.digest} の結果を表示`}
                        >
                          結果表示
                        </Button>
                      </TableCell>
                    </TableRow>
                  )}
                </For>
              </TableBody>
            </Table>
          </TableContainer>
        </Show>
      </Box>
    </Show>
  );
};
