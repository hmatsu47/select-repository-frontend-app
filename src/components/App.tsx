import { Component } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import { ImageList } from "./ImageList";
import { LastReleased } from "./LastReleased";
import { Message } from "./Message";
import { ReleaseSetting } from "./ReleaseSetting";
import { RepositorySelector } from "./RepositorySelector";
import { ServiceSelector } from "./ServiceSelector";
import { TitleBar } from "./TitleBar";

export const App: Component = () => {
  return (
    <>
      <TitleBar />
      <Box
        sx={{
          padding: "10px 10px 0 10px",
          width: "100%",
          minWidth: "1024px",
          display: "flex",
          // justifyContent: "center",
        }}
        aria-live="polite"
      >
        <Stack spacing={2} direction="column">
          <ServiceSelector />
          <LastReleased />
          <RepositorySelector />
          <ReleaseSetting />
          <Message />
          <ImageList />
        </Stack>
      </Box>
    </>
  );
};
