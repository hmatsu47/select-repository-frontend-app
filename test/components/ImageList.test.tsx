// @vitest-environment jsdom

import { describe, expect, test } from "vitest";
import "vi-fetch/setup";
import { fireEvent, render } from "solid-testing-library";
import { formatSnapshot } from "../common/formatSnapshot";
import { ImageList } from "../../src/components/ImageList";
import {
  imageUri,
  setImages,
  setImageUri,
  setReleaseAt,
  setRepository,
  setRepositoryUri,
} from "../../src/signal";
import { ImageItem } from "../../src/type";

describe("<ImageList />", () => {
  // 一覧表示されるケース
  const imageCheckList = [
    {
      title: "タグありの URI をセット",
      uri: "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest",
    },
    {
      title: "タグなしの URI をセット",
      uri: "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d",
    },
  ];
  imageCheckList.forEach((testCase) => {
    test(testCase.title, async () => {
      setImageUri(undefined);
      setReleaseAt(undefined);
      setRepository("hoge");
      setRepositoryUri(
        "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge"
      );
      setImages([
        {
          service: "service1",
          repository: "hoge",
          digest:
            "sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d",
          pushed_at: new Date(2022, 8, 12, 14, 45),
          repository_name: "hoge",
          size: 10000,
          tags: ["latest", "hogera"],
          uri: "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest",
        } as ImageItem,
        {
          service: "service1",
          repository: "hoge",
          digest:
            "sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d",
          pushed_at: new Date(2022, 8, 12, 14, 45),
          repository_name: "hoge",
          size: 2000000,
          tags: undefined,
          uri: "000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d",
        } as ImageItem,
      ]);
      const { container, findByText, findByTitle, unmount } = render(() => (
        <ImageList />
      ));
      const expectedListedUri = (await findByText(testCase.uri)) as HTMLElement;
      expect(expectedListedUri).toHaveTextContent(testCase.uri);
      // css の名前が動的に変わるので固定値に置換
      const htmlBefore = formatSnapshot(container.innerHTML);
      expect(htmlBefore).toMatchSnapshot();
      // ボタンクリック
      const button = (await findByTitle(
        `${testCase.uri} を選択`
      )) as HTMLInputElement;
      expect(button).toHaveTextContent("選択");
      fireEvent.click(button);
      expect(imageUri()).toBe(testCase.uri);
      // イメージ URI 指定後のスナップショット
      const htmlAfterUriSelect = formatSnapshot(container.innerHTML);
      expect(htmlAfterUriSelect).toMatchSnapshot();
      // 脆弱性スキャン結果表示リンクのテストは省略
      unmount();
    });
  });
  // 一覧表示されないケース（イメージなし）
  test("一覧未表示（イメージなし）", async () => {
    // すべて無指定
    setRepository("hoge");
    setRepositoryUri("000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge");
    setImages(null);
    const { container, findByText, unmount } = render(() => <ImageList />);
    const expected = (await findByText(
      "リポジトリにイメージがありません"
    )) as HTMLElement;
    // あえて expect は用意せず（完全一致しないので）
    // css の名前が動的に変わるので固定値に置換
    const html = formatSnapshot(container.innerHTML);
    expect(html).toMatchSnapshot();
    unmount();
  });
});
