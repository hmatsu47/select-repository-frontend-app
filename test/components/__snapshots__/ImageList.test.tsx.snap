// Vitest Snapshot v1

exports[`<ImageList /> > タグありの URI をセット 1`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table class=\\"MuiTable-root css-xxxxxxxx\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">選択</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">イメージURI<br>（リポジトリURI:タグ または リポジトリURI@ダイジェスト）</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">プッシュ日時</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">サイズ<br>（MB）</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">脆弱性調査<br>（要AWSログイン）</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest を選択\\" type=\\"button\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">0.01</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d の結果を表示\\" type=\\"button\\">結果表示<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d を選択\\" type=\\"button\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d の結果を表示\\" type=\\"button\\">結果表示<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></td></tr></tbody></table></div></div>"`;

exports[`<ImageList /> > タグありの URI をセット 2`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table class=\\"MuiTable-root css-xxxxxxxx\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">選択</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">イメージURI<br>（リポジトリURI:タグ または リポジトリURI@ダイジェスト）</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">プッシュ日時</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">サイズ<br>（MB）</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">脆弱性調査<br>（要AWSログイン）</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx_1\\" tabindex=\\"0\\" title=\\"000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest を選択\\" type=\\"button\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">0.01</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d の結果を表示\\" type=\\"button\\">結果表示<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d を選択\\" type=\\"button\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d の結果を表示\\" type=\\"button\\">結果表示<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></td></tr></tbody></table></div></div>"`;

exports[`<ImageList /> > タグなしの URI をセット 1`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table class=\\"MuiTable-root css-xxxxxxxx\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">選択</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">イメージURI<br>（リポジトリURI:タグ または リポジトリURI@ダイジェスト）</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">プッシュ日時</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">サイズ<br>（MB）</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">脆弱性調査<br>（要AWSログイン）</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest を選択\\" type=\\"button\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">0.01</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d の結果を表示\\" type=\\"button\\">結果表示<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d を選択\\" type=\\"button\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d の結果を表示\\" type=\\"button\\">結果表示<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></td></tr></tbody></table></div></div>"`;

exports[`<ImageList /> > タグなしの URI をセット 2`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table class=\\"MuiTable-root css-xxxxxxxx\\" aria-label=\\"simple table\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">選択</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">イメージURI<br>（リポジトリURI:タグ または リポジトリURI@ダイジェスト）</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">プッシュ日時</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">サイズ<br>（MB）</td><td class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"col\\">脆弱性調査<br>（要AWSログイン）</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest を選択\\" type=\\"button\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge:latest</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">0.01</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d の結果を表示\\" type=\\"button\\">結果表示<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\" scope=\\"row\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx_1\\" tabindex=\\"0\\" title=\\"000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d を選択\\" type=\\"button\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">000000000000.dkr.ecr.ap-northeast-1.amazonaws.com/hoge@sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" tabindex=\\"0\\" title=\\"sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d の結果を表示\\" type=\\"button\\">結果表示<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></td></tr></tbody></table></div></div>"`;

exports[`<ImageList /> > 一覧未表示（イメージなし） 1`] = `"<div class=\\"MuiBox-root css-xxxxxxxx\\" aria-live=\\"polite\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リポジトリにイメージがありません</h6></div>"`;
